### PR TITLE
Adds our-self-host

### PR DESCRIPTION
**Pull request by skttl**
Blogpost coming... :)
https://24days.in/umbraco-cms/2022/static-assets-taghelper/

### CODE REVIEW

**Comment by warrenbuckley** [comment]
@skttl just read blog post, mind fixing up the small conflict then can get this merged in :)

File: Our.Umbraco.TagHelpers/Extensions/WebHostEnvironmentExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
+﻿using Microsoft.AspNetCore.Hosting;
+using System;
+using System.IO;
+using Umbraco.Cms.Core;
+
+namespace Our.Umbraco.TagHelpers.Extensions
+{
+    [Obsolete("This should be removed, when the package gets upgraded past Umbraco 10")]
+    internal static class WebHostEnvironmentExtensions
```
Comment (warrenbuckley) [inline Our.Umbraco.TagHelpers/Extensions/WebHostEnvironmentExtensions.cs:9]: @skttl so if this was only for Umbraco 11 then we would need this class?

File: Our.Umbraco.TagHelpers/Services/SelfHostService.cs [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+
+                    selfHostedFile.Url = await GetSelfHostedUrl(selfHostedFile);
+                    return selfHostedFile;
+                }
+            });
+        }
+
+        private string GetFolderPath(Uri uri, string? subfolder = null)
+        {
+            var folderPath = _config["Our.Umbraco.TagHelpers.SelfHost.RootFolder"].IfNullOrWhiteSpace("~/assets"); ;
```
Comment (warrenbuckley) [inline Our.Umbraco.TagHelpers/Services/SelfHostService.cs:59]: Was the intention for it to be this key @skttl ?
Can we document it in the readme and perhaps align/sync it with the image one Andy Boot done.

**Comment by warrenbuckley** [comment]
Generally, looks all OK to me @skttl - just see the two small remarks.
Also do you think it's worth adding tests for this as well in the test project ?

File: Our.Umbraco.TagHelpers/Extensions/WebHostEnvironmentExtensions.cs (hunk 1 above)
Comment (skttl) [reply Our.Umbraco.TagHelpers/Extensions/WebHostEnvironmentExtensions.cs:9]: Yes :)

File: Our.Umbraco.TagHelpers/Services/SelfHostService.cs (hunk 2 above)
Comment (skttl) [reply Our.Umbraco.TagHelpers/Services/SelfHostService.cs]: Yes, but I've aligned with Andys now 👍

**Comment by skttl** [comment]
Added some tests, and fixed a bug - all good now :)

**Comment by warrenbuckley** [comment]
All good, sorry I took so long to get it merged @skttl 🙈

**Pull request merged**